services/wireguard/endpoint.go

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package wireguard
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+
+	"github.com/mdlayher/wireguardctrl"
+	"github.com/mdlayher/wireguardctrl/wgtypes"
+)
+
+// Provider is a configuration data required for establishing connection to the service provider.
+type Provider struct {
+	IP        net.IPNet
+	PublicKey wgtypes.Key
+	Endpoint  net.UDPAddr
+}
+
+// Consumer is a configuration data required to configure service consumer.
+type Consumer struct {
+	IP         net.IPNet
+	PrivateKey wgtypes.Key // TODO peer private key should be generated on consumer side
+}
+
+type connectionEndpoint struct {
+	iface    string
+	endpoint net.UDPAddr
+	wgClient *wireguardctrl.Client
+}
+
+// newConnectionEndpoint creates Wireguard client with predefined interface name and endpoint.
+func newConnectionEndpoint(iface string, endpoint net.UDPAddr) (*connectionEndpoint, error) {
+	wgClient, err := wireguardctrl.New()
```

Comment:
don't we want to pass this as an interface param to the newConnectionEndpoint? We could actually test this then

Reply:
`ConnectionEndpoint` is doing a lot of other low-level things like interface creation. We can move it to another abstraction layer, but in the end, we will have another layer that is hard to test using unit tests. 
I think it should be covered in E2E tests. We have a task for it and it will be done in another PR.